### PR TITLE
Fixed #23984 - Added Javascript i18n documentation

### PR DESCRIPTION
**Pull request by toastdriven**
This fleshes out the documentation around all of the exported
Javascript functions available from the ``javascript_catalog``
view.

### CODE REVIEW

File: docs/topics/i18n/translation.txt [hunk 1]
```diff
@@ -996,21 +996,38 @@ To use the catalog, just pull in the dynamically generated script like this:
...
+The ``gettext`` function behaves similarly to the standard ``gettext``
+interface within your Python code::
 
     document.write(gettext('this is to be translated'));
 
-There is also an ``ngettext`` interface::
+``ngettext``
+~~~~~~~~~~~~
+
+The ``ngettext`` function provides an interface pluralized words/phrases::
```
Comment (timgraham) [inline docs/topics/i18n/translation.txt]: "interface to pluralize"?
words/phrases -> words and phrases?

File: docs/topics/i18n/translation.txt [hunk 2]
```diff
@@ -996,21 +996,38 @@ To use the catalog, just pull in the dynamically generated script like this:
...
     s = ngettext('literal for the singular case',
             'literal for the plural case', object_cnt);
 
-and even a string interpolation function::
-
-    function interpolate(fmt, obj, named);
+``interpolate``
+~~~~~~~~~~~~~~~
 
+The ``interpolate`` function supports dynamically populating a format string::
```
Comment (timgraham) [inline docs/topics/i18n/translation.txt]: Should :: be a period?

File: docs/topics/i18n/translation.txt [hunk 3]
```diff
@@ -1043,6 +1060,91 @@ This isn't as fast as string interpolation in Python, so keep it to those
 cases where you really need it (for example, in conjunction with ``ngettext``
 to produce proper pluralizations).
 
+``get_format``
+~~~~~~~~~~~~~~
+
+The ``get_format`` has access to the configured i18n formatting settings and
+can lookup/return the format string for a given setting name::
```
Comment (timgraham) [inline docs/topics/i18n/translation.txt]: lookup and return?

File: docs/topics/i18n/translation.txt [hunk 4]
```diff
@@ -1043,6 +1060,91 @@ This isn't as fast as string interpolation in Python, so keep it to those
 cases where you really need it (for example, in conjunction with ``ngettext``
 to produce proper pluralizations).
 
+``get_format``
+~~~~~~~~~~~~~~
+
+The ``get_format`` has access to the configured i18n formatting settings and
```
Comment (timgraham) [inline docs/topics/i18n/translation.txt]: The get_format function

File: docs/topics/i18n/translation.txt [hunk 5]
```diff
@@ -1043,6 +1060,91 @@ This isn't as fast as string interpolation in Python, so keep it to those
...
+    document.write(gettext_noop('this will not be translated'));
+
+This is useful for stubbing out portions of the code that will need translation
+in the future.
+
+``pgettext``
+~~~~~~~~~~~~
+
+The ``pgettext`` function behaves like the Python variant
+(:func:`django.utils.translation.pgettext()`), providing a contextually
```
Comment (timgraham) [inline docs/topics/i18n/translation.txt]: Add ~ before django unless you think we need to render the entire path.

File: docs/topics/i18n/translation.txt [hunk 6]
```diff
@@ -1043,6 +1060,91 @@ This isn't as fast as string interpolation in Python, so keep it to those
...
+translated word::
+
+    document.write(pgettext('month name', 'May'));
+
+``npgettext``
+~~~~~~~~~~~~~
+
+The ``npgettext`` function also behaves like the Python variant
+(:func:`django.utils.translation.pgettext()`), providing a **pluralized**
```
Comment (timgraham) [inline docs/topics/i18n/translation.txt]: pgettext() -> npgettext()

File: docs/topics/i18n/translation.txt [hunk 7]
```diff
@@ -1043,6 +1060,91 @@ This isn't as fast as string interpolation in Python, so keep it to those
...
+
+    document.write(npgettext('group', 'party', 1));
+    // party
+    document.write(npgettext('group', 'party', 2));
+    // parties
+
+``pluralidx``
+~~~~~~~~~~~~~
+
+The ``pluralidx`` function works in a similar way to the ``pluralize`` template
```
Comment (timgraham) [inline docs/topics/i18n/translation.txt]: could make it a link: 
```
:tfilter:`plurlize`
```

**Comment by toastdriven** [comment]
@timgraham Thanks for the review! I think I fixed everything you pointed out.

**Comment by timgraham** [comment]
merged in 8ca9bc5ec326c0b01634dac264a9ba6f0857483e, thanks!

**Pull request closed without merging**